### PR TITLE
.editorconfig: various

### PR DESCRIPTION
**Pull request by zowoq**
~~Draft until we decide to keep the `editorconfig` CI check.~~

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -34,3 +34,9 @@ indent_size = 2
 # Match perl/python/shell scripts, set indent width of four
 [*.{pl,py,sh}]
 indent_size = 4
+
+# Disable for auto-generated hackage2nix file
+[pkgs/development/haskell-modules/hackage-packages.nix]
```
Comment (Mic92) [inline .editorconfig:39]: While you are at it, maybe also check if we need to add `pkgs/development/node-packages/node-packages-v*.nix`

**Comment by zowoq** [comment]
@Mic92 There is about 50 `*.upstream` files scattered around the repo, are they still in use or could they be removed?

I'm going through the repo by file type and size now looking for other outliers.

**Comment by Mic92** [comment]
I think mainly @7c6f434c uses them. I don't know what tool he uses it for.
I personally just use https://github.com/Mic92/nix-update/ which does not need extra meta information.

**Comment by zowoq** [comment]
> I personally just use https://github.com/Mic92/nix-update/ ...

I've been using as well. What do you think about adding it here? I send a PR it if you agree?

**Comment by Mic92** [comment]
> > I personally just use [Mic92/nix-update](https://github.com/Mic92/nix-update/) ...
> 
> I've been using as well. What do you think about adding it here? I send a PR it if you agree?

Feel free to add it with me as a maintainer.

**Comment by 7c6f434c** [comment]
`.upstream` files are usable with `update-walker` script which is inside Nixpkgs, and I do not always get around to updating based on them (but sometimes use them for updating).

File: .editorconfig (hunk 1 above)
Comment (zowoq) [reply .editorconfig:39]: Apart from not having final newlines they seem fine.

Some of the generated json files (`node-packages-*.json`, `recipes-archive-melpa.json`) will probably need to be excluded if we end up adding `json`.

Comment (zowoq) [reply .editorconfig:39]: Added `node-packages-v*.nix` and `gemset.nix`.

I guess we're probably going to end up doing this for a lot of the generated files.

**Comment by Mic92** [comment]
I think we can still merge this one.
It does not depend on the github ci check.

**Comment by zowoq** [comment]
I guess we should still try and get all of generated files listed and add some of the more common file types.

File: .editorconfig [hunk 2]
```diff
@@ -1,36 +1,59 @@
...
+indent_style = unset
+indent_size = unset
+trim_trailing_whitespace = unset
+
+# Disable for auto-generated node-packages files
+[pkgs/development/node-packages/node-packages-v*.nix]
+insert_final_newline = unset
+
+# Disable for auto-generated perl-packages file
+[pkgs/top-level/perl-packages.nix]
```
Comment (Mic92) [inline .editorconfig:58]: Some of the stuff might be automatic. Some of it is manual. But there might be a lot of errors still though given its size.

**Comment by zowoq** [comment]
I wanted to set a treewide default and require that files opt-out but there are too many miscellaneous files checked for a default to be enforced without it becoming tedious.

**Pull request merged**